### ISSUE TITLE
Chaturbate model goes offline shortly after starting capturing

### ISSUE THREAD

**Issue by Ptibouc77** [jrudess/mfc_cb.node#3]
Since some days i always get this when trying to record chaturbate model:

[20170126-195834] CB  modelname1 is now online, captured started
[20170126-195834] CB  modelname2 is now online, captured started
[20170126-195835] CB  Done, will search for new models in 300 second(s).
[20170126-195905] CB  modelname1  stopped streaming
[20170126-195905] CB  [ERROR] modelname1 : modelname1 _cb_20170126-195835.ts not found in capturing directory, cannot convert to mp4

But the model is still online, i also view that chaturbate have now two different type of stream "hlsSourceFast" and "hlsSourceSlow"

**Comment by jrudess** [comment]
I'm able to replicate the issue.

**Comment by jrudess** [comment]
I've also noticed that some models I add for CB are not even attempting to start capturing.   One theory is that the logic for building up the list of online models is now unreliably.   The first time it scans, it finds the model, and starts the capture.  Then the next time it scans it says the model is offline.

Sometime back I put in an extra check to stop ffmpeg if the model is offline to solve some rare cases where  the ffmpeg process did not properly end when a model disconnected.  That may be what is causing the capture to stop, but more testing is needed.

**Comment by jp2gmdxdxd** [comment]
I can report the same issue. Script is unusable. It generates no files.

[02/18/2017 - 18:23:31] Connecting to MyFreeCams chat server ychat32...
[02/18/2017 - 18:23:31] Login handshake completed. Logged in as 'Guest91047' with sessionId 407063591
[20170218-182336] MFC [DEBUG] Start searching for new models
[20170218-182336] MFC 1258 model(s) online
[20170218-182336] MFC [DEBUG] 5 model(s) in config
[20170218-182337] MFC AliciaCoxx is in public chat!
[20170218-182337] MFC LeightonBrook is in public chat!
[20170218-182337] MFC [DEBUG] 2 model(s) to capture
[20170218-182337] MFC AliciaCoxx, starting capture process
[20170218-182337] MFC LeightonBrook, starting capture process
[20170218-182337] MFC Done, will search for new models in 300 second(s).
[20170218-182337] MFC AliciaCoxx stopped streaming
[20170218-182337] MFC [ERROR] AliciaCoxx: AliciaCoxx_mfc_20170218-182337.ts not found in capturing directory, cannot convert to mp4
[20170218-182337] MFC LeightonBrook stopped streaming
[20170218-182337] MFC [ERROR] LeightonBrook: LeightonBrook_mfc_20170218-182337.ts not found in capturing directory, cannot convert to mp4
[20170218-182351] CB  2897 model(s) online
[20170218-182351] CB  [DEBUG] 2 model(s) in config
[20170218-182351] CB  Done, will search for new models in 300 second(s).
[20170218-182837] MFC [DEBUG] Start searching for new models
[20170218-182837] MFC 1305 model(s) online
[20170218-182837] MFC [DEBUG] 5 model(s) in config
[20170218-182837] MFC AliciaCoxx is in public chat!
[20170218-182837] MFC LeightonBrook is in public chat!
[20170218-182837] MFC [DEBUG] 2 model(s) to capture
[20170218-182837] MFC AliciaCoxx, starting capture process
[20170218-182837] MFC LeightonBrook, starting capture process
[20170218-182837] MFC Done, will search for new models in 300 second(s).
[20170218-182837] MFC LeightonBrook stopped streaming
[20170218-182837] MFC [ERROR] LeightonBrook: LeightonBrook_mfc_20170218-182837.ts not found in capturing directory, cannot convert to mp4
[20170218-182837] MFC AliciaCoxx stopped streaming
[20170218-182837] MFC [ERROR] AliciaCoxx: AliciaCoxx_mfc_20170218-182837.ts not found in capturing directory, cannot convert to mp4
[20170218-182851] CB  [DEBUG] Start searching for new models
[20170218-182912] CB  2865 model(s) online
[20170218-182912] CB  [DEBUG] 2 model(s) in config
[20170218-182912] CB  Done, will search for new models in 300 second(s).

**Comment by jrudess** [comment]
So far I have only had problems with CB and not with MFC. Did it work at one point with both for you?  Or are you getting this on the first try?

**Comment by jp2gmdxdxd** [comment]
I have been using capturebate-node for CB and it's working perfectly. For ex. now I'm recording valerymont_.

With your script I can't record anything :(
Fresh LOG:

[20170218-184643] CB  [DEBUG] Start searching for new models
[02/18/2017 - 18:46:43] Connecting to MyFreeCams chat server xchat59...
[02/18/2017 - 18:46:44] Login handshake completed. Logged in as 'Guest86728' with sessionId 407122811
[20170218-184649] MFC [DEBUG] Start searching for new models
[20170218-184649] MFC 1205 model(s) online
[20170218-184649] MFC [DEBUG] 5 model(s) in config
[20170218-184649] MFC AliciaCoxx is in public chat!
[20170218-184649] MFC LeightonBrook is in public chat!
[20170218-184649] MFC [DEBUG] 2 model(s) to capture
[20170218-184649] MFC AliciaCoxx, starting capture process
[20170218-184649] MFC LeightonBrook, starting capture process
[20170218-184650] MFC Done, will search for new models in 300 second(s).
[20170218-184650] MFC LeightonBrook stopped streaming
[20170218-184650] MFC AliciaCoxx stopped streaming
[20170218-184650] MFC [ERROR] LeightonBrook: LeightonBrook_mfc_20170218-184649.ts not found in capturing directory, cannot convert to mp4
[20170218-184650] MFC [ERROR] AliciaCoxx: AliciaCoxx_mfc_20170218-184649.ts not found in capturing directory, cannot convert to mp4
[20170218-184704] CB  2780 model(s) online
[20170218-184704] CB  [DEBUG] 3 model(s) in config
[20170218-184704] CB  [DEBUG] 1 model(s) to capture
[20170218-184704] CB  valerymont_ is now online, captured started
[20170218-184704] CB  Done, will search for new models in 300 second(s).
[20170218-184704] CB  valerymont_ stopped streaming
[20170218-184704] CB  [ERROR] valerymont_: valerymont__cb_20170218-184704.ts not found in capturing directory, cannot convert to mp4

**Comment by jp2gmdxdxd** [comment]
Here is my config.yml

captureDirectory: capturing
completeDirectory: captured
modelScanInterval: 300
autoConvertType: mp4
autoDelete: false
dateFormat: YYYYMMDD-HHmmss
debug: true
modelcolor: magenta
timecolor: grey
sitecolor: green
debugcolor: yellow
errorcolor: red
includeSiteInFile: true
enableMFC: true
enableCB: true
mfcmodels:
  - 3623822
  - 12513225
  - MsKate_
  - AliciaCoxx
  - LeightonBrook
cbmodels:
  - aynmarie
  - theemilygrey
  - valerymont_

and updates.yml

includeMfcModels: []
excludeMfcModels: []
includeCbModels: []
excludeCbModels: []

**Comment by jrudess** [comment]
Based on the timestamps, this actually looks like ffmpeg may be exiting with an error immediately.  That would trigger this exact behavior.  In main.js, uncomment line 358:

`  //common.dbgMsg(site, 'Launching ffmpeg ' + spawnArgs)`

This will print out the ffmpeg command line being used.  Replace all the commas with spaces, and then try running that ffmpeg command line and see if any errors are reported.

**Comment by jrudess** [comment]
You might also remove the -fatal from the ffmpeg command line so that more info is printed.

**Comment by jrudess** [comment]
Just noticed that you have some issues in your config.yml file.   The mfc models section only allows the MFC model ID numbers, and not their names.   To add an mfc model you have to put them into updates.yml like this:

`includeMfcModels: [MissKate_, AliciaCoxx]`

**Comment by jp2gmdxdxd** [comment]
Still not working. I commented line 358 and changed updates.yml, but still norhing ...

LOG:

[20170218-191047] CB  [DEBUG] Start searching for new models
[02/18/2017 - 19:10:47] Connecting to MyFreeCams chat server xchat48...
[02/18/2017 - 19:10:48] Login handshake completed. Logged in as 'Guest87464' with sessionId 407184243
[20170218-191052] MFC [DEBUG] Start searching for new models
[20170218-191052] MFC 1071 model(s) online
[20170218-191052] MFC [DEBUG] 3 model(s) in config
[20170218-191052] MFC AliciaCoxx is in public chat!
[20170218-191052] MFC [DEBUG] 1 model(s) to capture
[20170218-191052] MFC AliciaCoxx, starting capture process
[20170218-191052] MFC [DEBUG] Launching ffmpeg -hide_banner,-v,fatal,-i,http://video471.myfreecams.com:1935/NxServer/ngrp:mfc_109651949.f4v_mobile/playlist.m3u8,-c,copy,/root/mfc1/mfc_cb.node/capturing/AliciaCoxx_mfc_20170218-191052.ts
[20170218-191052] MFC Done, will search for new models in 300 second(s).
[20170218-191053] MFC AliciaCoxx stopped streaming
[20170218-191053] MFC [ERROR] AliciaCoxx: AliciaCoxx_mfc_20170218-191052.ts not found in capturing directory, cannot convert to mp4
[20170218-191105] CB  2598 model(s) online
[20170218-191105] CB  [DEBUG] 3 model(s) in config
[20170218-191105] CB  [DEBUG] 1 model(s) to capture
[20170218-191105] CB  valerymont_ is now online, captured started
[20170218-191106] CB  [DEBUG] Launching ffmpeg -hide_banner,-v,fatal,-i,https://edge60.stream.highwebmedia.com/live-edge/valerymont_-sd-63e6aa155b8bdc7ad430cf18b0ef9b6ebdf24525652a34b017ccaf74ead3a836_fast_aac/playlist.m3u8,-c,copy,/root/mfc1/mfc_cb.node/capturing/valerymont__cb_20170218-191106.ts
[20170218-191106] CB  Done, will search for new models in 300 second(s).
[20170218-191106] CB  valerymont_ stopped streaming
[20170218-191106] CB  [ERROR] valerymont_: valerymont__cb_20170218-191106.ts not found in capturing directory, cannot convert to mp4

**Comment by jrudess** [comment]
Please try manually running ffmpeg using the command line printed above.  Hopefully an error will be printed out from ffmpeg with more useful info.   

Replace the commas in the printed command line with spaces, and remove -fatal.

**Comment by jp2gmdxdxd** [comment]
There is no files downloaded. What should I do exactly ?

**Comment by jrudess** [comment]
Does ffmpeg print any errors?   Can you paste the ffmpeg output to http://pastebin.com/ and send the link?

**Comment by jp2gmdxdxd** [comment]
Dude, I don't understand you. What I should do? There are no files downloaded from MFC or CB.

**Comment by jp2gmdxdxd** [comment]
[20170218-215814] CB  [DEBUG] Start searching for new models
[02/18/2017 - 21:58:14] Connecting to MyFreeCams chat server ychat32...
[02/18/2017 - 21:58:15] Login handshake completed. Logged in as 'Guest15451' wit                                               h sessionId 407457541
[20170218-215820] MFC [DEBUG] Start searching for new models
[20170218-215820] MFC 1210 model(s) online
[20170218-215820] MFC [DEBUG] 2 model(s) in config
[20170218-215820] MFC 1 model(s) to include
[20170218-215820] MFC AriannaSecret added to capture list
[20170218-215820] MFC [DEBUG] Rewriting config.yml
[20170218-215821] MFC AriannaSecret is in public chat!
[20170218-215821] MFC [DEBUG] 1 model(s) to capture
[20170218-215821] MFC AriannaSecret, starting capture process
[20170218-215821] MFC Done, will search for new models in 300 second(s).
[20170218-215821] MFC [DEBUG] { Error: spawn ffmpeg ENOENT
    at exports._errnoException (util.js:1023:11)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:193:32)
    at onErrorNT (internal/child_process.js:359:16)
    at _combinedTickCallback (internal/process/next_tick.js:74:11)
    at process._tickCallback (internal/process/next_tick.js:98:9)
  code: 'ENOENT',
  errno: 'ENOENT',
  syscall: 'spawn ffmpeg',
  path: 'ffmpeg',
  spawnargs:
   [ '-hide_banner',
     '-v',
     'fatal',
     '-i',
     'http://video474.myfreecams.com:1935/NxServer/ngrp:mfc_116915028.f4v_mobile                                               /playlist.m3u8',
     '-c',
     'copy',
     'capturing/AriannaSecret_mfc_20170218-215821.ts' ] }
[20170218-215821] MFC AriannaSecret stopped streaming
[20170218-215821] MFC [ERROR] AriannaSecret: AriannaSecret_mfc_20170218-215821.t                                               s not found in capturing directory, cannot convert to mp4
[20170218-215838] CB  2940 model(s) online
[20170218-215838] CB  [DEBUG] 3 model(s) in config
[20170218-215838] CB  Done, will search for new models in 300 second(s).

**Comment by januszPawlacz2** [comment]
I have the same issue.

[20170327-200004] CB  [DEBUG] 4 model(s) in config
[20170327-200004] CB  [DEBUG] 1 model(s) to capture
[20170327-200004] CB  sarahadams is now online, captured started
[20170327-200004] CB  Done, will search for new models in 300 second(s).
[20170327-200005] CB  sarahadams stopped streaming
[20170327-200005] CB  [ERROR] sarahadams: sarahadams_cb_20170327-200004.ts not found in capturing directory, cannot convert to mp4

**Comment by Farquaad56** [comment]
same issue here with the last release.
chaturbate record doesn't work

**Comment by jrudess** [comment]
Got some info from some other folks to ignore the edge part of the URL. For example, edge13 server is pretty bad and causes lots of disconnects, while edge35 is good. This may depend on where you are located.

Plan right now is to add a new config.yml option. When set, it will specify a list of edge* servers to choose from. When not set, it will just use the server provided by CB.

**Comment by snickity** [comment]
I have the same issue with a specific model on MFC.

[20170513_164525] MFC Bekkabella is in public chat!
[20170513_164525] MFC Bekkabella recording started (Bekkabella_mfc_20170513_164525.ts)
[20170513_165039] MFC [ERROR] Bekkabella, Bekkabella_mfc_20170513_164525.ts not found in capturing directory, cannot convert to mp4
[20170513_165526] MFC Bekkabella recording started (Bekkabella_mfc_20170513_165526.ts)
[20170513_170040] MFC [ERROR] Bekkabella, Bekkabella_mfc_20170513_165526.ts not found in capturing directory, cannot convert to mp4
[20170513_170526] MFC Bekkabella recording started (Bekkabella_mfc_20170513_170526.ts)
[20170513_171041] MFC [ERROR] Bekkabella, Bekkabella_mfc_20170513_170526.ts not found in capturing directory, cannot convert to mp4
[20170513_171527] MFC Bekkabella recording started (Bekkabella_mfc_20170513_1715 27.ts)
[20170513_172041] MFC [ERROR] Bekkabella, Bekkabella_mfc_20170513_171527.ts not found in capturing directory, cannot convert to mp4
[20170513_172528] MFC Bekkabella recording started (Bekkabella_mfc_20170513_172528.ts)
[20170513_173042] MFC [ERROR] Bekkabella, Bekkabella_mfc_20170513_172528.ts not found in capturing directory, cannot convert to mp4
[20170513_173529] MFC Bekkabella's cam is off.
[20170513_180532] MFC Bekkabella has logged off.

**Comment by jrudess** [comment]
Some improvements to CB flows have been pushed to the development branch, and in my local testing CB appears to be much more reliable now.   This could use some broader testing before release, so if you have time and can download the dev branch and try it out, that would be appreciated.

**Comment by jrudess** [comment]
I believe this problem is now resolved with v0.4.   If anyone still see this problem with that release, please update this sighting with debug logs and I will reopen it.

**Comment by Skyline2k** [comment]
Sry, I'm still seeing this with both the dev and master branch.
```
[2017-06-20_15.51.27] CB  14 model(s) in config
[2017-06-20_15.51.30] CB  fullacum123 is in public chat!
[2017-06-20_15.51.30] CB  [DEBUG] 1 model(s) to capture
[2017-06-20_15.51.31] CB  fullacum123 recording started (fullacum123_2017-06-20_15.51.31.ts)
[2017-06-20_15.51.31] CB  [DEBUG] Done, waiting 30 seconds.
[2017-06-20_15.51.32] CB  [ERROR] fullacum123, fullacum123_2017-06-20_15.51.31.ts not found in capturing directory, cannot convert to mkv
[2017-06-20_15.52.05] CB  [DEBUG] 1 model(s) to capture
[2017-06-20_15.52.06] CB  fullacum123 recording started (fullacum123_2017-06-20_15.52.06.ts)
[2017-06-20_15.52.06] CB  [DEBUG] Done, waiting 30 seconds.
[2017-06-20_15.52.06] CB  [ERROR] fullacum123, fullacum123_2017-06-20_15.52.06.ts not found in capturing directory, cannot convert to mkv
```
